lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -38,6 +38,7 @@ class KubernetesResource
     LAST_APPLIED_ANNOTATION = "kubectl.kubernetes.io/last-applied-configuration"
     SENSITIVE_TEMPLATE_CONTENT = false
     SERVER_DRY_RUNNABLE = false
+    PREFETCH_KINDS_ON_SYNC = []
```

Comment:
Nit: This doesn't match the method name you chose, and kinda sounds like it would make the sync do this automatically. Suggestion: `SYNC_DEPENDENCIES` or `PREWARM_KINDS_FOR_SYNC`.